tesseract/tesseract_planning/tesseract_process_managers/include/tesseract_process_managers/core/process_planning_server.h

```diff
@@ -116,6 +128,17 @@ class ProcessPlanningServer
    * @return A future to monitor progress
    */
   std::future<void> run(tf::Taskflow& taskflow);
+#endif  // SWIG
+
+#ifdef SWIG
+  %extend
+  {
+    tesseract_planning::ProcessPlanningFuture* run(const ProcessPlanningRequest& request)
+    {
+      return new tesseract_planning::ProcessPlanningFuture(std::move($self->run(request)));
+    }
```

Comment:
Just curious, why the need to extend and return a pointer? Who deletes it?

Reply:
The class is move-only, and SWIG will copy a return by value. Returning a pointer fixes the problem. I think SWIG should delete it but I will check.

Reply:
This was an error, SWIG would not delete the object in this context since the ownership flag wasn't set. I changed it to a shared_ptr return so it is deleted properly.